src/ert/run_models/base_run_model.py

```diff
@@ -495,6 +496,60 @@ def send_snapshot_event(self, event: CloudEvent) -> None:
                 )
             )
 
+    async def run_monitor(self, ee_config: EvaluatorServerConfig) -> bool:
+        try:
+            event_logger.debug("connecting to new monitor...")
+            async with MonitorAsync(ee_config.get_connection_info()) as monitor:
+                event_logger.debug("connected")
+                async for event in monitor.track():
+                    if event["type"] in (
+                        EVTYPE_EE_SNAPSHOT,
+                        EVTYPE_EE_SNAPSHOT_UPDATE,
+                    ):
+                        self.send_snapshot_event(event)
+                        if event.data.get(STATUS) in [
+                            ENSEMBLE_STATE_STOPPED,
+                            ENSEMBLE_STATE_FAILED,
+                        ]:
+                            event_logger.debug(
+                                "observed evaluation stopped event, signal done"
+                            )
+                            await monitor.signal_done()
+                            # should_exit = True
+                        if event.data.get(STATUS) == ENSEMBLE_STATE_CANCELLED:
+                            event_logger.debug(
+                                "observed evaluation cancelled event, exit drainer"
+                            )
+                            # Allow track() to emit an EndEvent.
+                            return False
+                    elif event["type"] == EVTYPE_EE_TERMINATED:
+                        event_logger.debug("got terminator event")
+
+                    if not self._end_queue.empty():
+                        event_logger.debug("Run model canceled - during evaluation")
+                        self._end_queue.get()
+                        await monitor.signal_cancel()
+                        event_logger.debug(
+                            "Run model canceled - during evaluation - cancel sent"
+                        )
+            # This sleep needs to be there. Refer to issue #1250: `Authority
+            # on information about evaluations/experiments`
+            # time.sleep(self._next_ensemble_evaluator_wait_time)
+        except (ConnectionRefusedError, ClientError) as e:
+            if not self.isFinished():
+                event_logger.debug(f"connection refused: {e}")
+        except ConnectionClosedError as e:
+            # The monitor connection closed unexpectedly
+            event_logger.debug(f"connection closed error: {e}")
+        except BaseException:
+            event_logger.exception("unexpected error: ")
```

Comment:
Does logger.exception include the stack trace of the exception?

Reply:
This one was just taken from before / main. I'll improve logging once ee_async comes into play.